### ISSUE TITLE
GitHub Actions ?

### ISSUE THREAD

**Issue by MSP-Greg** [ruby/openssl#277]
I've got GH Actions running in my fork, one file, Ubuntu macOS, & MinGW.

Ubuntu is all OpenSSL 1.1.1, macOS is 1.0.2, and MinGW is 1.0.2 & 1.1.1.

Ubuntu 16.04 - Ruby 2.3
Ubuntu 18.04 - Ruby 2.4 thru 2.6
Windows - Ruby 2.4 thru 2.6

All 1.0.2 jobs are passing, but all 1.1.1 jobs are failing.  Same test that some of the master jobs are failing on.

MinGW builds need an update to extconf.rb to compile with the current MSYS2 system.

If you'd like the file...

**Comment by ioquatix** [comment]
Can you make a PR.

**Comment by MSP-Greg** [comment]
I thought #278 would fix the issue on 1.1.1.  Now, the Ubuntu jobs are passing, but MinGW still has a test failure shown below.  I'll do the PR:
```
1) Failure: test_finished_messages(OpenSSL::TestSSL):
  exceptions on 1 threads:
  #<Thread:0x00000000043e65a8@D:/a/ruby-openssl/ruby-openssl/test/utils.rb:261 dead>:
  D:/a/ruby-openssl/ruby-openssl/lib/openssl/ssl.rb:491:in `accept': An established connection was aborted by the software in your host machine. - SSL_accept (Errno::ECONNABORTED)
  	from D:/a/ruby-openssl/ruby-openssl/lib/openssl/ssl.rb:491:in `accept'
  	from D:/a/ruby-openssl/ruby-openssl/test/utils.rb:234:in `block (3 levels) in start_server'
  	from D:/a/ruby-openssl/ruby-openssl/test/utils.rb:230:in `loop'
  	from D:/a/ruby-openssl/ruby-openssl/test/utils.rb:230:in `block (2 levels) in start_server'
D:/a/ruby-openssl/ruby-openssl/test/envutil.rb:281:in `assert_join_threads'
D:/a/ruby-openssl/ruby-openssl/test/utils.rb:289:in `block in start_server'
D:/a/ruby-openssl/ruby-openssl/test/utils.rb:203:in `pipe'
D:/a/ruby-openssl/ruby-openssl/test/utils.rb:203:in `start_server'
D:/a/ruby-openssl/ruby-openssl/test/test_ssl.rb:433:in `test_finished_messages'
     430:     client_finished = nil
     431:     client_peer_finished = nil
     432: 
  => 433:     start_server(accept_proc: proc { |server|
     434:       server_finished = server.finished_message
     435:       server_peer_finished = server.peer_finished_message
     436:     }) { |port|
```

**Comment by MSP-Greg** [comment]
See PR https://github.com/ruby/openssl/pull/279.

Above issue fixed with 3rd commit.
